<a href="https://colab.research.google.com/github/faznafathima942/Assignments/blob/main/ASSIGNMENT_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **ASSIGNMENT 3**

## Dataset: StudentsPerformance.csv



---







### IMPORTING NECESSARY LIBRARIES

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### READ THE DATA

In [2]:
data=pd.read_csv(r"/content/drive/MyDrive/StudentsPerformance.csv")
data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

### **1.  To find out how many males and females participated in the test:**
We can use the value_counts() function to count the number of occurrences of each unique value in the 'gender' column. 

In [3]:
data['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

Inference: There were 518 female students and 482 male students who participated in the test.

### **2. To analyze the students' parental level of education:**

We can use the value_counts() function to count the number of occurrences of each unique value in the 'parental level of education' column.

In [4]:
data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

Inference: The majority of the students' parents have completed some college education.

### **3. To find out who scores the most on average for math, reading, and writing based on Gender and Test preparation course:**

We can use the groupby() function to group the data by gender and test preparation course and then use the mean() function to calculate the average scores for math, reading, and writing.

In [5]:
data.groupby(['gender', 'test preparation course'])[['math score', 'reading score', 'writing score']].mean()

math score  reading score  writing score
gender test preparation course                                          
female completed                 67.108696      77.375000      78.793478
       none                      61.320359      69.955090      68.982036
male   completed                 72.988506      70.793103      70.339080
       none                      66.467532      62.581169      59.551948

Inference: 

Female students who completed the test preparation course had the highest average scores for reading, and writing, and those who didnot completed scored the highest in reading and writing 

male students who completed the test preparation course had the highest average math score, and males who did not complete the test preparation course score the highest in math.

### **4. To analyze the scoring variation for math, reading, and writing based on Gender and Test preparation course:**

We can use the groupby() function to group the data by gender and test preparation course and then use the std() function to calculate the standard deviation for math, reading, and writing.

In [6]:
data.groupby(['gender', 'test preparation course'])[['math score', 'reading score', 'writing score']].std()

math score  reading score  writing score
gender test preparation course                                          
female completed                 14.428233      12.695493      12.240788
       none                      16.510686      14.636981      15.016497
male   completed                 14.039164      13.612420      12.887910
       none                      14.335746      13.595718      13.468284

Inference: There is more variation in the scores of female students who did not complete the test preparation course compared to those who did. There is relatively less variation in the scores of male students.

### **5. To give bonus points to the top 25% of students based on their math score**

we can do this by calculating the 25th percentile using the "quantile()" function of pandas library on the "math score" column of the dataset. Then,by filtering the dataset to keep only those students who have scored above the 25th percentile in math, and then by assigning bonus points to them.

In [7]:
# Calculate the 25th percentile of math score
math_percentile = data['math score'].quantile(0.75)

# Filter the dataset to keep only those students who scored above the 25th percentile in math
top_students = data[data['math score'] >= math_percentile]

# Assign bonus points to the top 25% of students based on their math score
top_students['bonus points'] = 5

# Print the top 5 rows of the dataset with bonus points assigned
print(top_students.head())


     gender race/ethnicity parental level of education         lunch  \
414  female        group A                 high school  free/reduced   
415  female        group B             master's degree  free/reduced   
416  female        group B           bachelor's degree  free/reduced   
417  female        group B             master's degree      standard   
418  female        group B                 high school      standard   

    test preparation course  math score  reading score  writing score  \
414               completed          77             88             85   
415               completed          77             97             94   
416                    none          77             85             87   
417                    none          77             90             84   
418               completed          77             82             89   

     bonus points  
414             5  
415             5  
416             5  
417             5  
418             5  


<ipython-input-7-c92f32cced80>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  top_students['bonus points'] = 5


 The top 25% of students based on their math score have been assigned 5 bonus points.



---



---

